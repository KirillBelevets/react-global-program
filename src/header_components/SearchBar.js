import React from 'react'
import './header.scss'
import SearchButton from "./SearchButton";


const searchBar = () => {

    return (
        <div className="search-bar-wrapper">
            <div className="search-bar-label text-white uppercase">
                Find your movie
            </div>
            <div className="search-bar flex">
                <input type="text" className="text-grey-2" placeholder="What do you want to watch?"/>
                <SearchButton />
            </div>
        </div>
    )
}

export default searchBar