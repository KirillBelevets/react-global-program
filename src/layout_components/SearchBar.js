import React, {useState} from 'react'
import './header.scss'
import {useHistory} from "react-router-dom";


const SearchBar = () => {
    const [searchValue, setSearchValue] = useState('')
    const history = useHistory()

    const searchHandler = (e) => {
        let val = e.target.value
        setSearchValue(val)
    }

    const submitHandler = () => {
        if (searchValue.length > 0) {
            history.push(`/search/${searchValue}`)
        }
    }

    return (
        <div className="search-bar-wrapper">
            <div className="search-bar-label text-white text-uppercase">
                Find your movie
            </div>
            <div className="search-bar display-flex">
                <input type="text" className="text-grey-2" placeholder="What do you want to watch?"
                       onChange={searchHandler}/>
                <div>
                    <button type="submit" className="search-button text-uppercase cursor-pointer"
                            onClick={submitHandler}>Search
                    </button>
                </div>
            </div>
        </div>
    )
}

export default SearchBar
