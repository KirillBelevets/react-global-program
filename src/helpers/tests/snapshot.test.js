import React from "react"
import { render } from '@testing-library/react'

import SearchBar from '../../layout_components/SearchBar'

describe('App', () => {
    test('renders App snapshot', () => {
        const { asFragment } = render(<SearchBar/>)

        expect(asFragment(<SearchBar/>)).toMatchSnapshot()
    })
})
