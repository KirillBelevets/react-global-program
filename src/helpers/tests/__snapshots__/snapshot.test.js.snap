// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-bar-wrapper"
  >
    <div
      class="search-bar-label text-white text-uppercase"
    >
      Find your movie
    </div>
    <div
      class="search-bar display-flex"
    >
      <input
        class="text-grey-2"
        placeholder="What do you want to watch?"
        type="text"
      />
      <div>
        <button
          class="search-button text-uppercase cursor-pointer"
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
